### PR TITLE
ensure mib-name passed as parameter

### PR DESCRIPTION
**Pull request by barryodonovan**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

File: includes/polling/functions.inc.php [hunk 1]
```diff
@@ -386,16 +386,12 @@ function poll_mib_def($device, $mib_name_table, $mib_subdir, $mib_oids, $mib_gra
...
 
-    if (is_null($rrd_name)) {
-        if (stristr($mib_name_table, 'UBNT')) {
-            list($mib,) = explode(':', $mib_name_table, 2);
-            $measurement_name = strtolower($mib);
-        } else {
-            list($mib,$file) = explode(':', $mib_name_table, 2);
-            $measurement_name = strtolower($file);
-        }
+    if (stristr($mib_name_table, 'UBNT')) {
```
Comment (murrant) [inline includes/polling/functions.inc.php]: we have a function called str_contains(), please use that.

**Comment by murrant** [comment]
This code seems confusing to me, can it just be simplified to (including the section below):
```
    if (is_null($rrd_name)) {
        list($mib,$file) = explode(':', $mib_name_table, 2);
        if (str_contains($mib_name_table, 'UBNT', true)) {
            $rrd_name = strtolower($mib);
        } else {
             $rrd_name = strtolower($file);
        }
    }
```

**Comment by barryodonovan** [comment]
Aaarrrgghhh!

I can't understand what is happening!

The only change need to be a parameter, rrd_name =null and then an if rrd_name == null, rrd_name = measurement_name

For some reason the code is correct on my dev box but not on github...

Regarding the str_contains, that's the original code, I'm all in favor of cleaning it up, but that's how it was merged last year... Then again, I'm also in favor of native functions where available...

Barry

**Comment by murrant** [comment]
strstr() does not always behave as expected.  The function simply extracts a string from another string.   

The reason we added that function is because we have strops() strstr() and their case insensitive variants all over the place inconsistently.

As for the code it is a very complicated piece of code with too many temp variables just to set $rrd_name to one of three values.

**Comment by barryodonovan** [comment]
Fair enough, good points.

I'll give it all another shot today.

________________________________

**Pull request merged**